Broken Jar

The `jar-with-dependencies` does not work anymore. 

Bug reproduction steps:
```
git clone git@github.com:SpoonLabs/sonarqube-repair.git
cd sonarqube-repair
./init.sh
mvn package -DskipTests
java -jar target/sonarqube-repair-1.1-SNAPSHOT-jar-with-dependencies.jar --originalFilesPath src/test/resources/ --rulesKey 2116
```

Error output:
```
WARNING: sun.reflect.Reflection.getCallerClass is not supported. This will impact performance.
Exception in thread "main" java.lang.ExceptionInInitializerError
	at sonarquberepair.DefaultRepair.repair(DefaultRepair.java:33)
	at sonarquberepair.Main.start(Main.java:84)
	at sonarquberepair.Main.main(Main.java:90)
Caused by: java.lang.UnsupportedOperationException: No class provided, and an appropriate one cannot be found.
	at org.apache.logging.log4j.LogManager.callerClass(LogManager.java:576)
	at org.apache.logging.log4j.LogManager.getLogger(LogManager.java:601)
	at org.apache.logging.log4j.LogManager.getLogger(LogManager.java:588)
	at spoon.Launcher.<clinit>(Launcher.java:589)
	... 3 more
```

On first sight it could very possibly be a dependency collision between `LogManager` of `sonarjava` and `spoon-core`  by checking line `at spoon.Launcher.<clinit>(Launcher.java:589)` . Since `spoon-core` use log4j , its [LogManager](https://logging.apache.org/log4j/2.x/log4j-api/apidocs/org/apache/logging/log4j/LogManager.html) can use the method `LogManager.getLogger()` without any arguments, while `sonarjava` use slf4j , its [LogManager](http://www.slf4j.org/api/org/apache/log4j/LogManager.html) need at least 1 argument causing the error above.

WDYT ? @fermadeiral , @monperrus  :)

Here's the new error output:
```
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
Exception in thread "main" java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at sonarquberepair.DefaultRepair.repair(DefaultRepair.java:51)
	at sonarquberepair.Main.start(Main.java:84)
	at sonarquberepair.Main.main(Main.java:90)
Caused by: java.lang.NoClassDefFoundError: org/sonar/java/checks/CastArithmeticOperandCheck
	at sonarquberepair.processor.CastArithmeticOperandProcessor.<init>(CastArithmeticOperandProcessor.java:19)
	... 7 more
Caused by: java.lang.ClassNotFoundException: org.sonar.java.checks.CastArithmeticOperandCheck
	at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:352)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
	... 8 more

```

--- a/.travis.yml
+++ b/.travis.yml
@@ -5,3 +5,7 @@ jdk:
 
 before_install:
   - ./init.sh
+
+script:
+  - java -jar /home/travis/build/SpoonLabs/sonarqube-repair/target/sonarqube-repair-1.1-SNAPSHOT-jar-with-dependencies.jar --originalFilesPath /home/travis/build/SpoonLabs/sonarqube-repair/src/test/resources/ArrayHashCodeAndToString.java --ruleKeys 2184
+  - mvn test

--- a/pom.xml
+++ b/pom.xml
@@ -36,7 +36,7 @@
         <dependency>
             <groupId>fr.inria.gforge.spoon</groupId>
             <artifactId>spoon-core</artifactId>
-            <version>8.2.0-beta-2</version>
+            <version>8.1.0-beta-2</version>
         </dependency>
         <!-- https://mvnrepository.com/artifact/org.sonarsource.java/java-checks-testkit -->
         <dependency>
@@ -53,8 +53,7 @@
         <dependency>
             <groupId>org.sonarsource.java</groupId>
             <artifactId>sonar-java-plugin</artifactId>
-            <version>5.14.0.18788</version>
-            <scope>provided</scope>
+            <version>5.14.0-SNAPSHOT</version>
         </dependency>
         <dependency>
             <groupId>com.martiansoftware</groupId>

